### PR TITLE
Fix Ubuntu arm build break.

### LINKED ISSUE #23268: broken arm ubuntu build from the recent update
https://github.com/dotnet/coreclr/pull/23249  broke arm32 Ubuntu Jenkins testing (at least), see example [here](https://ci.dot.net/job/dotnet_coreclr/job/master/job/arm_cross_checked_ubuntu_crossgen_comparison_prtest).

@AaronRobinsonMSFT could you please fix it and explain why you merged PR with failed jobs?

cc @dotnet/arm32-contrib

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Add platformdefines.h include to fix build break.

Fixes #23268

### CODE REVIEW

**Comment by AaronRobinsonMSFT** [comment]
@jkoritzinsky Thanks. Actually submit this one as well. If the PR passes before the other we can merge this one instead. I would prefer for this to be the fix since it is what I will do tomorrow anyways.

**Comment by jkoritzinsky** [comment]
@dotnet-bot test this please

File: tests/src/Common/hostpolicymock/HostpolicyMock.cpp [hunk 1]
```diff
@@ -6,6 +6,7 @@
 // Used for testing CoreCLR/Corlib functionality which calls into hostpolicy.
 
 #include <string>
+#include "platformdefines.h"
```
Comment (AaronRobinsonMSFT) [inline tests/src/Common/hostpolicymock/HostpolicyMock.cpp]: should be `<platformdefines.h>`?

**Comment by AaronRobinsonMSFT** [comment]
Let the race begin! - https://github.com/dotnet/coreclr/pull/23269

File: tests/src/Common/hostpolicymock/HostpolicyMock.cpp (hunk 1 above)
Comment (jkoritzinsky) [reply tests/src/Common/hostpolicymock/HostpolicyMock.cpp]: My personal preference is to only do `<>` for system headers, but I'll go change it.

**Comment by AaronRobinsonMSFT** [comment]
All builds pass in Azure piplines. OSX release passed without issues, but OSX debug had an issue in the pipline due to logs.

>Artifact build_OSX_x64_checked_Logs already exists for build 124917.

This is the preferable fix so I am commiting this one in lieu of https://github.com/dotnet/coreclr/pull/23269.

**Pull request merged**